specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2021-03-01-preview/EntityQueries.json

```diff
@@ -0,0 +1,651 @@
+    },
+    "ActivityEntityQuery": {
+      "description": "Represents Activity entity query.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/EntityQuery"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/ActivityEntityQueriesProperties",
+          "description": "Activity entity query properties",
+          "x-ms-client-flatten": true
+        }
+      },
+      "type": "object",
+      "x-ms-discriminator-value": "Activity"
+    },
+    "ActivityCustomEntityQuery": {
+      "description": "Represents Activity entity query.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/CustomEntityQuery"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/ActivityEntityQueriesProperties",
+          "description": "Activity entity query properties",
+          "x-ms-client-flatten": true
+        }
+      },
+      "type": "object",
+      "x-ms-discriminator-value": "Activity"
+    },
+    "ActivityEntityQueriesProperties": {
+      "description": "Describes activity entity query properties",
+      "properties": {
+        "title": {
+          "description": "The entity query title",
+          "type": "string"
+        },
+        "content": {
+          "description": "The entity query content to display in timeline",
+          "type": "string"
+        },
+        "description": {
+          "description": "The entity query description",
+          "type": "string"
+        },
+        "queryDefinitions": {
+          "description": "The Activity query definitions",
+          "properties": {
+            "query": {
+              "description": "The Activity query to run on a given entity",
+              "type": "string"
+            }
+          },
+          "type": "object"
+        },
+        "inputEntityType": {
+          "$ref": "#/definitions/EntityInnerType",
+          "description": "The type of the query's source entity"
+        },
+        "requiredInputFieldsSets": {
+          "description": "List of the fields of the source entity that are required to run the query",
+          "items": {
+            "description": "Sub sets of the field of the source entity that are required to run the query",
+            "items": {
+              "description": "Required input field name",
+              "type": "string"
+            },
+            "type": "array"
+          },
+          "type": "array"
+        },
+        "entitiesFilter": {
+          "description": "The query applied only to entities matching to all filters",
+          "type": "object"
+        },
+        "templateName": {
+          "description": "The template id this activity was created from",
+          "type": "string"
+        },
+        "enabled": {
+          "description": "Determines whether this activity is enabled or disabled.",
+          "type": "boolean"
+        },
+        "createdTimeUtc": {
```

Comment:
created and lastModified should only be handled via the common systemData contract.  https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-contracts.md#system-metadata-for-all-azure-resources

Reply:
Do I need to implement all "systemData" properties?
"systemData":{
      "createdBy": "<string>",
      "createdByType": "<User|Application|ManagedIdentity|Key>",
      "createdAt": "<date-time>",
      "lastModifiedBy": "<string>",
      "lastModifiedByType": "<User|Application|ManagedIdentity|Key>",
      "lastModifiedAt": "<date-time>"
  },

or just have the createdAt and lastModifiedAt under the systemData property?

Reply:
The entire contract needs to be implemented.  More info is in the wiki:  https://armwiki.azurewebsites.net/api_contracts/ResourceSystemData.html?q=systemdata

Reply:
Done

Reply:
/OK,
and what changes do I need to apply?
besides removing the Create and LastModeified in the "ActivityEntityQueriesProperties"?
do I need to declare this "systemData" in the spec?

Reply:
yes, it goes at the root of your resource next to "id", "name", and "type".  There is a common definition you can use called `systemData` here:  specification\common-types\resource-management\v2\types.json

Reply:
Hi @pilor, 
I have talked with our privacy champion, he has raised the concern regarding the use identifiers included in the SystemData. 
why do we need to store and manage user identifiers if they are not required in our scenario?
this will change the resource tagging, because today we do not store any user identifiers in the DB.
We only need the Created\LastModified time.

is it possible to simply return the user identifiers without storing\managing them? to avoid this "user identification" tagging?

Reply:
@jennyhunter-msft, can you address the concerns about systemData?

Reply:
@pilor, @jennyhunter-msft  - any updates regarding this?
thank you

Reply:
@pilor, @jennyhunter-msft , have some questions.
is this API supported in FFX and AGC? 
we feel uncomfortable to use this new API, and we have our API up and running for private customers for the past 6 months, so we prefer to have this Created and LastUpdated as is for the moment.

Reply:
systemData is a standard contract supported in all clouds that over time will provide a consistent createdBy/lastUpdateBy experience for all resources.  Each resource should not be exposing similar information in its own way.

Reply:
I understand he motivation to align all resources to this convention.
currently, all our resources are managing these properties themselves.
When we started this feature private preview, we were not aware of this "SystemData" API.
We will use it in the future, so we can align to ARM standards, but now, changing the existing implementation is not an option.
we have an API which is up and running, and we simply want to correct our swagger to support and expose it to the public.
Thank you

Reply:
I'd suggest removing those properties from this api-version so its not as big of a change for your consumers when you add systemData in the next api-version.  I'll leave that decision to you

Reply:
Thank you @pilor, this is much appreciated.
What else we have for closing this one?

Reply:
Hi @pilor , @jennyhunter-msft,
What are the next steps towards merging this one?
I still have another 2 PRs I need to conduct